### PR TITLE
Profiles: clear current registration name when RegisterENSNavigator unmounts

### PR DESCRIPTION
**Pull request by benisgold**
Fixes RNBW-3369

## What changed (plus any additional context for devs)
- clear current registration name when RegisterENSNavigator unmounts

## PoW (screenshots / screen recordings)
https://user-images.githubusercontent.com/15272675/164755392-9214db61-aafa-4fa5-a6f6-22fe29c1178f.mp4

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Review by estebanmino — APPROVED** [review]

**Comment by jxom** [comment]
Does [this line](https://github.com/rainbow-me/rainbow/pull/3244/files#diff-0dcee63cce4aba213c0a39671f24b20234afbbc300d0f82b80c4fef2ec93bf9eR133) not work?

**Comment by benisgold** [comment]
no it doesn't @jxom i'll see if i can get that working

**Comment by benisgold** [comment]
@jxom it works when you dismiss by sliding down the screen, but not when you click cancel

**Comment by jxom** [comment]
Ah okay. We probably need to get it to work for cancel then.

```tsx
navigation.emit({
            target: Routes.REGISTER_ENS_NAVIGATOR,
            type: 'dismiss',
          });
```

I think that works.

File: src/navigation/RegisterENSNavigator.js [hunk 1]
```diff
@@ -113,23 +111,31 @@ export default function RegisterENSNavigator() {
...
       removeRecordByKey('avatar');
       setAccentColor(colors.purple);
       clearValues();
       clearCurrentRegistrationName();
-    };
-    navigation.addListener('dismiss', dismiss);
-    return () => navigation.removeListener('dismiss', dismiss);
-  });
+    },
+    [
```
Comment (jxom) [inline src/navigation/RegisterENSNavigator.js:131]: thank you for this lol

**Review by jxom — APPROVED** [review]

**Pull request merged**